File: insertionSort.py
def insertionSort(array): 
    for i in range(1, len(array)):
        j = i
        while j > 0 & array[j] < array[j - 1]:
            swap(j, j - 1, array)
            j -= 1
    return array

def swap(idx1, idx2, array):
    array[idx1], array[idx2] = array[idx2], array[idx1]

File: insertionSort.js
const insertionSort = (array) => {
    const arrayLen = array.length
    for (let i = 0; i < arrayLen; i++) {
        let j = i
        while (j > 0 && array[j] < array[j - 1]) {
            let temp = array[j]
            array[j] = array[j - 1]
            array[j - 1] = temp
            j--
        }
    }
    return array
}